source/pdo_sqlsrv/pdo_dbh.cpp

```diff
@@ -98,6 +102,24 @@ struct pdo_txn_isolation_conn_attr_func
     static void func( connection_option const* /*option*/, zval* value_z, sqlsrv_conn* conn, std::string& /*conn_str*/ TSRMLS_DC );
 };
 
+#ifdef _WIN32
+struct pdo_int_conn_str_func {
+
+    static void func( connection_option const* option, zval* value, sqlsrv_conn* /*conn*/, std::string& conn_str TSRMLS_DC )
+    {
+        TSRMLS_C;
+        SQLSRV_ASSERT( Z_TYPE_P( value ) == IS_STRING, "Wrong zval type for this keyword" ) 
```

Comment:
Well, an integer is expected for it, but it's a string in the connection string. This function is based on pdo_int_conn_attr_func::func immediately below, which is used for handling the login timeout, also an integer, and uses a similar assert. In conn.cpp, the login timeout is handled as an integer in the zval, so I did it that way. Why the difference between the sqlsrv and pdo source, I don't know.